### ISSUE TITLE
project generator crashes if run in Remote Desktop window

### ISSUE THREAD

**Issue by steven-chith** [microsoftarchive/openFrameworks#1]
Repro on Windows 8.1 Professional.

**Comment by ghost** [comment]
lower priority- try to repro again
separate issue: should winrt continue to have a separate project generator?

**Comment by bilderbuchi** [comment]
> should winrt continue to have a separate project generator?

not from our side, no. @ofZach any news in the meantime about the PG restructuring?

**Comment by ofZach** [comment]
no all project generators should be the same.   This means changes to how we parse addons, etc, get reflected on all platforms.  This was the issue before, where we had a variety of scripts across all platforms which got out of sync with each other and had a variety of non standard features. 
arturo and I discussed a plan to restructure common parts of the PG into their own addon post 0.8.1 -- it was too much to do before 0.8.1, but we'll start as a top priority for 0.8.2.  At any rate, we should maybe jump on a call or email exchange to discuss what winRT needs here?

**Comment by stammen** [comment]
The generator code for WinRT (and later on Windows Phone, WP8) may not to be too different from the Win32 version. However, the important thing to consider is that WinRT and WP8 are separate platforms from Win32. The project file format is different, they require keys, manifests and asset folders, etc. So we will need to have Win32, WinRT and WP8 project generators.  This is probably similar to iOS and Mac being different platforms.

**Comment by bilderbuchi** [comment]
How does this necessitate having a separate application for generating the files, though? We currently have one application generating all our project files across 5+ platforms (afaik).

**Comment by ofZach** [comment]
yes - it should be possible, unless these files are not xml based, to just extend the base classes in project generator to do what you want. 

this class is really an important starting point: 
https://github.com/ofZach/projectGeneratorSimple/blob/master/src/projects/baseProject.h

there's different implementations, but basically, you need to write an implementation that loads an existing project (from the scripts folder), parses it and modifies it, and handles what ever external files it needs to copy.    It's typically all XML and it's a fair amount of xpath expressions (we use pugixml) for this.

**Comment by stammen** [comment]
Our plan is to merge our project generator code with yours. There should be no need for a separate generator.

**Comment by stammen** [comment]
The project files for WinRT are XML based so this should work fine. Thanks for the guidance.
